### DISCUSSION TITLE
How to toggle -p (--prune) on for git fetch by default?

### DISCUSSION THREAD

**Discussion by DivineDominion** [magit/magit#4684] [Support] [answered]
Magit always starts a fetch (`f`) with the --prune option disabled, even though 

    git config --global fetch.prune true

and 

    git config remote.origin.prune true

are both enabled. Is there another way to make Magit prune by default?

Command line git SO question for reference: https://stackoverflow.com/questions/18308535/automatic-prune-with-git-fetch-or-pull

**Comment by DivineDominion** [comment, marked as answer]
Ah, might answer my own question: https://magit.vc/manual/magit/Customizing.html#Customizing

>  Certain settings like fetch.prune=true are respected by Magit commands (because they simply call the respective Git command) but their value is not reflected in the respective transient buffers. In this case the --prune argument in magit-fetch might be active or inactive, but that doesn’t keep the Git variable from being honored by the suffix commands anyway. So pruning might happen despite the --prune arguments being displayed in a way that seems to indicate that no pruning will happen. 

Ok, so it's only the transient menu that doesn't highlight the setting.

Would be cool if it did for consistency, but it's a stretch goal I guess :)

**Reply by tarsius** [reply]
> Would be cool if it did for consistency, but it's a stretch goal I guess :)

Yes: #2993. :grinning:

**Reply by DivineDominion** [reply]
Also, linked from that over-arching enhancement: #2610 as a detail
